From 758495bf32da1eee0bb9c19a4f5481a108e54adb Mon Sep 17 00:00:00 2001
From: Christina Jacob <cjacob@marvell.com>
Date: Thu, 13 Jun 2019 14:30:37 +0530
Subject: [PATCH 295/386] octeontx2-af: Update fwadata structure with few more
 reserved fields.

Updates fwdata with the fields missed in the atf commit
03d997c10c6c9db3ed64a8b3839cbb11b4a75633.
The reserved fields ensure that the correctness of
fw_data is not affected evenif the
definitions in atf and kernel are out of sync.

Change-Id: Ie68aab8c43b305c3953a9e312d31bc3df0139509
Signed-off-by: Christina Jacob <cjacob@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/11247
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/mbox.h | 3 +++
 drivers/net/ethernet/marvell/octeontx2/af/rvu.h  | 1 -
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
index 65552f33fb5c..91cb0a953e51 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/mbox.h
@@ -473,6 +473,7 @@ struct sfp_eeprom_s {
 #define SFP_EEPROM_SIZE 256
 	u16 sff_id;
 	u8 buf[SFP_EEPROM_SIZE];
+	u64 reserved;
 };
 
 enum fec_type {
@@ -491,6 +492,8 @@ struct cgx_lmac_fwdata_s {
 	u64 advertised_link_modes;
 	/* Only applicable if SFP/QSFP slot is present */
 	struct sfp_eeprom_s sfp_eeprom;
+#define RESERVED_MEM 1024
+	u64 reserved[RESERVED_MEM];
 };
 
 struct cgx_fw_data {
diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
index 49dcfc15a95f..336bd856ccf6 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu.h
@@ -386,7 +386,6 @@ struct rvu_fwdata {
 	u64 rclk;
 	u64 mcam_addr;
 	u64 mcam_sz;
-#define RESERVED_MEM 1024
 	u64 reserved[RESERVED_MEM];
 #define CGX_MAX         3
 #define CGX_LMACS_MAX   4
-- 
2.17.1

